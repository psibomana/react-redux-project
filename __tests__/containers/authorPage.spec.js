import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import configureStore from 'redux-mock-store'
import AuthorPage from '../../src/containers/authorPage'

describe('Author Page', () => {
  let store, wrapper, initialState;
  beforeEach(() => {
    const mockStore = configureStore();
    initialState = {
      AuthorReducer: {
        authors: [
          {
            id: 'cory-house',
            firstName: 'Cory',
            lastName: 'House'
          },
          {
            id: 'scott-allen',
            firstName: 'Scott',
            lastName: 'Allen'
          },
          {
            id: 'dan-wahlin',
            firstName: 'Dan',
            lastName: 'Wahlin'
          }
        ]
      }
    };
    store = mockStore(initialState);
    wrapper = shallow(<AuthorPage store={store} />);
  })

  it('should render as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('should map state to props', () => {
    expect(wrapper.props().authors.length)
      .toEqual(initialState.AuthorReducer.authors.length)
  })
});
