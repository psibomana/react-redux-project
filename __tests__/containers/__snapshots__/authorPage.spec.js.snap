// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Page should render as expected 1`] = `
<AuthorPage
  actions={
    Object {
      "addAuthor": [Function],
      "deleteAuthor": [Function],
      "getAuthor": [Function],
      "listAuthors": [Function],
      "updateAuthor": [Function],
    }
  }
  authors={
    Array [
      Object {
        "firstName": "Cory",
        "id": "cory-house",
        "lastName": "House",
      },
      Object {
        "firstName": "Scott",
        "id": "scott-allen",
        "lastName": "Allen",
      },
      Object {
        "firstName": "Dan",
        "id": "dan-wahlin",
        "lastName": "Wahlin",
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
